components/eventing-controller/pkg/backend/nats/jetstream/jetstream_test.go

```diff
@@ -1476,3 +1477,103 @@ func TestSubscriptionSubjectIdentifierNamespacedName(t *testing.T) {
 		})
 	}
 }
+
+// TestJetStream_NoNATSSubscription tests if the error is being triggered
+// when expected in-memory representation of NATS subscriptions are missing.
+func TestJetStream_NoNATSSubscription(t *testing.T) {
+	// given
+	testEnvironment := setupTestEnvironment(t)
+	jsBackend := testEnvironment.jsBackend
+	defer testEnvironment.natsServer.Shutdown()
+	defer testEnvironment.jsClient.natsConn.Close()
+	initErr := jsBackend.Initialize(nil)
+	require.NoError(t, initErr)
+
+	// create New Subscriber
+	subscriber := evtesting.NewSubscriber()
+	defer subscriber.Shutdown()
+	require.True(t, subscriber.IsRunning())
+
+	testCases := []struct {
+		name                            string
+		subOpts                         []evtesting.SubscriptionOpt
+		givenManuallyDeleteSubscription bool
```

Comment:
instead of this, cant we check the len of `givenFilterToDelete` and delete it?

Reply:
but it's two additional variables for a check in one testcase. But it can stay as is if you like